test/functional/eval/capture_spec.lua

```diff
@@ -0,0 +1,39 @@
+local helpers = require('test.functional.helpers')
+local eq = helpers.eq
+local eval = helpers.eval
+local clear = helpers.clear
+local execute = helpers.execute
+local source = helpers.source
+
+describe('capture()', function()
+  before_each(clear)
+
+  it('returns the same result with :redir', function()
+    execute('redir => g:foo | messages | redir END')
+    eq(eval('g:foo'), eval('capture(":messages")'))
+  end)
+
+  it('returns the output of the commands if the argument is List', function()
+    eq("foobar", eval([[capture(['echon "foo"', 'echon "bar"'])]]))
```

Comment:
Also `funcs.capture({'echon "foo"', 'echon "bar"'})` and you do not have to bother with third level of escaping. (Basically I would not suggest to use `eval` here at all, in all tests that are currently present or were suggested by me `funcs.capture()` works just fine or `eval()` cannot be used because `exc_exec`/`redir_exec` is needed.)